webapp/src/ts/services/db.service.ts

```diff
@@ -16,19 +16,18 @@ import { POUCHDB_OPTIONS } from '../constants';
   providedIn: 'root'
 })
 export class DbService {
-  private cache = {}
+  private cache = {};
   private isOnlineOnly;
 
   constructor(private session:SessionService, private location:LocationService) {
 
-
     this.isOnlineOnly = this.session.isOnlineOnly();
-
     if (!this.isOnlineOnly) {
       // delay the cleanup so it's out of the main startup sequence
+      const self = this;
```

Comment:
I'd wish to avoid this notation, if at all possible. 
Is it necessary here?

Reply:
In many cases yes it was needed, specially when two nested functions need to access to the `this` reference`, but not sure in this case, I'll check.

Reply:
If you use arrow functions, it shouldn't be needed, which seems to be the case here.